controllers/helpers.go

```diff
@@ -253,7 +265,7 @@ func newCloudProviderConfig(d azure.ClusterScoper) (controlPlaneConfig *CloudPro
 			SecurityGroupName:            subnet.SecurityGroup.Name,
 			SecurityGroupResourceGroup:   d.Vnet().ResourceGroup,
 			Location:                     d.Location(),
-			VMType:                       "vmss",
+			VMType:                       "vm",
```

Comment:
Maybe I missed something, could you provide more info for this change? More specifically:
1. Will this cloudProviderConfig be applied to the azure.json file?
2. Besides Cloud Controller Manager, is there any other component using this cloud config?
3. I remember Cloud Contoller Manager expects the vmType to be set to "standard" rather than "vm" for non-vmss cases. Do you set it to "vm" for other purpose? Just want to double check in case it does not align with what Cloud Controller Mananger expects :)

Reply:
This change came about because we noticed CAPZ *always* sets `VMType: vmss`, but doesn't need to if MachinePools aren't even available.

1. Yes, this is the config that is the azure.json file.
2. I've been told there are other components that might read the config, but I'm not clear which--maybe cluster-autoscaler?
3. This is just a bug resulting from me not re-checking the provider docs, I intended to use `standard`, not `vm`. I'll update the PR soon.

Reply:
@zmyzheng I see now in the cloud-provider-azure docs that the `vmType` values have changed:

| Name | Description | Remark |
|---|---|---|
| vmType | The type of azure nodes. Candidate values are: `vmss`, `vmssflex` and `standard` | Optional, default to `standard` |
| enableVmssFlexNodes | Enable supporting for VMSS Flex virtual machines in vmss cluster. It should only be used when vmType is “vmss” | Since v1.26.0, default is false |

I'm confused about what this new `vmType: vmssflex` value is and how it interacts with `enableVmssFlexNodes`.

Reply:
This section of the docs explains it better: https://cloud-provider-azure.sigs.k8s.io/topics/node-types/

I actually think the main doc that I excerpted above is misleading and should be changed. I'll make a PR.

So based on CAPZ's usage patterns, we would never use `vmType: vmssflex` because we won't have a cluster that is 100% VMSS Flexible nodes. 

My only question is if this PR is worthwhile: if we know a cluster is 100% standalone or Availability Set nodes, is there any advantage to using `vmType: standard`?